# 1.2 Displacement and $LCB$ of a sailing_yacht
![](../figures/yacht.png)

In [2]:
import pandas as pd
from mms185 import simpson
import matplotlib.pyplot as plt
import numpy as np

Sectional areas of a vessel at a constant interval along the length (d=1 m) are as follows:

In [6]:
df = pd.DataFrame()
area=[0.00, 0.21, 0.50, 0.71, 0.82, 0.79, 0.68, 0.15, 0.08, 0.02, 0.00]
df['ordinate'] = np.arange(len(area))
df['area [m2]'] = area
df

ordinate  area [m2]
0          0       0.00
1          1       0.21
2          2       0.50
3          3       0.71
4          4       0.82
5          5       0.79
6          6       0.68
7          7       0.15
8          8       0.08
9          9       0.02
10        10       0.00

## Find the vessels displacement ($\Delta$) in tonnes (saltwater)
The displacement in tonnes can be calculated as the volume $\nabla$ times the density $\rho$ of salt water (1.025 tonnes/m3)

$$ \Delta=\nabla \cdot \rho $$

The volume can be calculated as the integral of the section areas along the length of the ship:

$$ V = \int_{AP}^{FP} A \cdot dx $$

Using simspon integration that yields:
$$ \frac{2d}{3} \sum_{n=0}^{10} S/M \cdot A $$

In [9]:
simpson.simpson_table(y=df['area [m2]'])

y  S/M  product
0   0.00  0.5     0.00
1   0.21  2.0     0.42
2   0.50  1.0     0.50
3   0.71  2.0     1.42
4   0.82  1.0     0.82
5   0.79  2.0     1.58
6   0.68  1.0     0.68
7   0.15  2.0     0.30
8   0.08  1.0     0.08
9   0.02  2.0     0.04
10  0.00  0.5     0.00

In [13]:
d=1
V = simpson.simpson(y=df['area [m2]'],d=d)
V

3.893333333333333

The volume displacement $\nabla$ of the yacht is 3.89 $m^3$

In [12]:
rho=1.025
M = rho*V
M

3.990666666666666

The displacement is 4 tonnes in salt water.

## Find the longitudinal centre of buoyancy ($LCB$) from $AP$.

In [14]:
N = len(df)
df['x'] = df['ordinate']*d

In [18]:
y = df['area [m2]']*df['x']
My = simpson.simpson(y=y,d=d)
My/V


4.116438356164383

The longitudinal centre of buoyancy ($LCB$) is 4.12 m from $AP$.